aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1728,22 +1728,29 @@ QuantizerPtr create_subtensor_quantizer(const Tensor& self, bool is_select, int6
   return quantizer;
 }
 
-Tensor select(const Tensor& self, int64_t dim, int64_t index_) {
+Tensor select(const Tensor& self, int64_t dim, int64_t index) {
+  return at::select_symint(self, dim, c10::SymInt{index});
+}
+
+Tensor select(const Tensor& self, Dimname dim, int64_t index) {
+  return at::select_symint(self, dimname_to_position(self, dim), c10::SymInt{index});
```

Comment:
I'll accept this but it is technically a perf regression!

Reply:
Why is this a perf regression?

Reply:
you're converting to SymInt and doing compute there, which entails extra tests to see if it's symbolic or not. Not a big deal; it's more annoying when there's a symintarrayref input